common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -131,9 +135,26 @@ public boolean isCancellable() {
         return result == null;
     }
 
+    private static class LeanCancellationException extends CancellationException {
+        private static final long serialVersionUID = 2794674970981187807L;
+
+        @Override
+        public Throwable fillInStackTrace() {
+            setStackTrace(CANCELLATION_STACK);
+            return this;
+        }
+        @Override
+        public String toString() {
+            return CancellationException.class.getName();
+        }
+    }
+
     @Override
     public Throwable cause() {
         Object result = this.result;
+        if (result == CANCELLATION_CAUSE_HOLDER) {
+            throw new LeanCancellationException();
```

Comment:
this is not correct... I would have hoped we have unit tests for this :). This method should return the `cause()` and not throw it.

Reply:
@njhill please add a unit test for this :)

Reply:
LOL oops I think this was part of a last min refactor, the `throw` was meant to be `return` of course... will fix and add unit test